vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java

```diff
@@ -0,0 +1,1060 @@
+     *            maximum column width back to the default value.
+     */
+    public void setMaximumColumnWidth(String maxColWidth) {
+        getStyle().set("--vaadin-dashboard-col-max-width", maxColWidth);
+    }
+
+    /**
+     * Returns the minimum row height of the dashboard.
+     *
+     * @return the minimum row height of the dashboard
+     */
+    public String getMinimumRowHeight() {
+        return getStyle().get("--vaadin-dashboard-row-min-height");
+    }
+
+    /**
+     * Sets the minimum row height of the dashboard.
+     *
+     * @param minRowHeight
+     *            the new minimum row height. Pass in {@code null} to set the
+     *            minimum row height back to the default value.
+     */
+    public void setMinimumRowHeight(String minRowHeight) {
+        getStyle().set("--vaadin-dashboard-row-min-height", minRowHeight);
+    }
+
+    /**
+     * Returns the spacing of the dashboard. This value adjusts the spacing
+     * between elements within the component and the space around its outer
+     * edges.
+     *
+     * @return the spacing of the dashboard
+     */
+    public String getSpacing() {
+        return getStyle().get("--vaadin-dashboard-spacing");
+    }
+
+    /**
+     * Sets the spacing of the dashboard. This value adjusts the spacing between
+     * elements within the component and the space around its outer edges.
+     *
+     * @param spacing
+     *            the new spacing. Pass in {@code null} to set the spacing back
+     *            to the default value.
+     */
+    public void setSpacing(String spacing) {
+        getStyle().set("--vaadin-dashboard-spacing", spacing);
+    }
+
+    /**
+     * Sets the dashboard editable.
+     *
+     * @param editable
+     *            whether to set the dashboard editable
+     */
+    public void setEditable(boolean editable) {
+        getElement().setProperty("editable", editable);
+    }
+
+    /**
+     * Returns whether the dashboard is editable.
+     *
+     * @return whether the dashboard is editable
+     */
+    public boolean isEditable() {
+        return getElement().getProperty("editable", false);
+    }
+
+    /**
+     * Sets the dashboard layout dense.
+     *
+     * @param dense
+     *            whether to set the dashboard layout dense
+     */
+    public void setDenseLayout(boolean dense) {
+        getElement().setProperty("denseLayout", dense);
+    }
+
+    /**
+     * Returns whether the dashboard layout is dense.
+     *
+     * @return whether the dashboard layout is dense
+     */
+    public boolean isDenseLayout() {
+        return getElement().getProperty("denseLayout", false);
+    }
+
+    /**
+     * Adds an item moved listener to this dashboard.
+     *
+     * @param listener
+     *            the listener to add, not <code>null</code>
+     * @return a handle that can be used for removing the listener
+     */
+    public Registration addItemMovedListener(
```

Comment:
How do you move a widget or section programmatically?

Reply:
Currently, the way to do it would be using add/remove/get methods in combination. Programatically reordering items after initializing them probably is not that common of a use case. Do you have an suggestion for such an API? Pinging @tomivirkki @yuriy-fix @rolfsmeds for visibility.

Reply:
It looks like you could move a widget, but you can not move a section because there is no way to insert it at an index.

Reply:
It's still possible but is a little cumbersome and would require a few steps:
- getting the children
- reordering them
- removing all from dashboard
- adding them one by one again (which also needs casting)

Reply:
OK, could be added later I guess.